# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1260476772 28800
#      Thu Dec 10 12:26:12 2009 -0800
# Node ID dbbec33d14f8cea25dad333afeda7ca7c44c0bd1
# Parent  5f3c10ecb150d758ba784177aa3b909763ae24ef
Final patch to enable autotools VPATH builds of documentation

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-09  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: Copy DOCSTRINGS from srcdir to builddir if it does not
+	exist.  Enables VPATH builds.
+
 2009-12-08  Petr Mikulik <mikulik@physics.muni.cz>
 
 	* plot/gnuplot_drawnow.m: Prune prompt lines from dumb terminal output.
 	* plot/__go_draw_figure__.m: Add missing 'unset multiplot' to restore
 	hotkey and mouse functionality for 2D plots.
 
 2009-12-09  Rik <octave@nomad.inbox5.com>
 
diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -324,16 +324,17 @@ EXTRA_DIST += \
   $(IMAGES) \
   $(FCN_FILES) \
   $(FCN_FILES_IN) \
   $(GEN_FCN_FILES)
 
 all-local: $(GEN_FCN_FILES) $(PKG_ADD_FILES) DOCSTRINGS
 
 DOCSTRINGS: mkdoc $(FCN_FILES) $(GEN_FCN_FILES) gethelp$(BUILD_EXEEXT) 
+	@cp -a --update $(srcdir)/$@ $@ ; exit 0;
 	$(srcdir)/mkdoc "$(srcdir)" $(FCN_FILES) -- $(GEN_FCN_FILES) > $@-t
 	$(simple_move_if_change_rule)
 
 gethelp$(BUILD_EXEEXT): gethelp.cc
 	$(BUILD_CXX) $(BUILD_CXXFLAGS) -o $@ $^ $(BUILD_LDFLAGS)
 
 $(GEN_FCN_FILES) : %.m : %.in Makefile
 	$(do_subst_config_vals)
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-09  Rik <octave@nomad.inbox5.com>
+
+	* Makefile.am: Copy DOCSTRINGS from srcdir to builddir if it does not
+	exist.  Enables VPATH builds.
+
 2009-12-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-dld-fcn.h (octave_dld_function::get_shlib): New method.
 	* ov-mex-fcn.h (octave_mex_function::get_shlib): New method.
 	* defun.cc (get_current_shlib): New function.
 	* defun-int.h: Declare it.
 	(octave_auto_shlib): New class.
 
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -623,16 +623,17 @@ graphics-props.cc: graphics.h.in genprop
 	$(AWK) -v emit_graphics_props=1 -f $(srcdir)/genprops.awk $< > $@-t
 	mv $@-t $@
 
 DLD-FUNCTIONS/PKG_ADD: $(DLD_DYNAMIC_DEF_FILES) mk-pkg-add
 	$(srcdir)/mk-pkg-add $(DLD_DYNAMIC_DEF_FILES) > $@-t
 	mv $@-t $@
 
 DOCSTRINGS: gendoc$(BUILD_EXEEXT)
+	@cp -a --update $(srcdir)/$@ $@ ; exit 0;
 	./gendoc > $@-t
 	$(simple_move_if_change_rule)
 
 doc-files: $(DEF_FILES)
 	echo $(DEF_FILES) > $@-t
 	mv $@-t $@
 
 gendoc.cc: doc-files mkgendoc
